gitlab/resource_gitlab_topic_test.go

```diff
@@ -0,0 +1,160 @@
+package gitlab
+
+import (
+	"fmt"
+	"strconv"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/xanzy/go-gitlab"
+)
+
+func TestAccGitlabTopic_basic(t *testing.T) {
+	var topic gitlab.Topic
+	rInt := acctest.RandInt()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckGitlabTopicDestroy,
+		Steps: []resource.TestStep{
+			// Create a topic with default options
+			{
+				Config: testAccGitlabTopicRequiredConfig(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGitlabTopicExists("gitlab_topic.foo", &topic),
+					testAccCheckGitlabTopicAttributes(&topic, &testAccGitlabTopicExpectedAttributes{
+						Name: fmt.Sprintf("foo-req-%d", rInt),
+					}),
+				),
+			},
+			// Update the topics values
+			{
+				Config: testAccGitlabTopicFullConfig(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGitlabTopicExists("gitlab_topic.foo", &topic),
+					testAccCheckGitlabTopicAttributes(&topic, &testAccGitlabTopicExpectedAttributes{
+						Name:        fmt.Sprintf("foo-full-%d", rInt),
+						Description: "Terraform acceptance tests",
+					}),
+				),
+			},
+			// Update the topics values back to their initial state
+			{
+				Config: testAccGitlabTopicRequiredConfig(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGitlabTopicExists("gitlab_topic.foo", &topic),
+					testAccCheckGitlabTopicAttributes(&topic, &testAccGitlabTopicExpectedAttributes{
+						Name: fmt.Sprintf("foo-req-%d", rInt),
+					}),
+				),
+			},
+			// Updating the topic to have a description before it is deleted
+			{
+				Config: testAccGitlabTopicFullConfig(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGitlabTopicExists("gitlab_topic.foo", &topic),
+					testAccCheckGitlabTopicAttributes(&topic, &testAccGitlabTopicExpectedAttributes{
+						Name:        fmt.Sprintf("foo-full-%d", rInt),
+						Description: "Terraform acceptance tests",
+					}),
+				),
+			},
+			// Verify import
```

Comment:
Would be nice to have a separate `_import` test here, instead of "just" a TestStep.

Reply:
Shoving the import test into a final TestStep actually comes from [this lint rule](https://github.com/bflad/tfproviderlint/tree/main/passes/AT002) and [a recommendation in the Terraform docs](https://www.terraform.io/plugin/sdkv2/resources/import#recommendations-for-import)
> The TestStep including ImportState testing should not be performed solely in a separate acceptance test. This duplicates testing infrastructure/time and does not check that all resource configurations import into Terraform properly.

Reply:
Oh, didn't know that! Thanks!